includes/Core/Authentication/Authentication.php

```diff
@@ -149,29 +149,16 @@ public function __construct(
 		User_Options $user_options = null,
 		Transients $transients = null
 	) {
-		$this->context = $context;
-
-		if ( ! $options ) {
-			$options = new Options( $this->context );
-		}
-		$this->options = $options;
-
-		if ( ! $user_options ) {
-			$user_options = new User_Options( $this->context );
-		}
-		$this->user_options = $user_options;
-
-		if ( ! $transients ) {
-			$transients = new Transients( $this->context );
-		}
-		$this->transients = $transients;
-
+		$this->context           = $context;
+		$this->options           = $options ?: new Options( $this->context );
+		$this->user_options      = $user_options ?: new User_Options( $this->context );
+		$this->transients        = $transients ?: new Transients( $this->context );
 		$this->google_proxy      = new Google_Proxy( $this->context );
 		$this->credentials       = new Credentials( new Encrypted_Options( $this->options ) );
 		$this->verification      = new Verification( $this->user_options );
-		$this->verification_meta = new Verification_Meta( $this->user_options, $this->transients );
+		$this->verification_meta = new Verification_Meta( $this->user_options );
 		$this->verification_file = new Verification_File( $this->user_options );
-		$this->profile           = new Profile( $user_options );
+		$this->profile           = new Profile( $this->user_options );
 		$this->first_admin       = new First_Admin( $this->options );
 	}
```

Comment:
I believe the new `User_Setting` instances here still need to be registered (in `Authentication::register()` alongside `Credentials`).